genie-web/src/main/java/com/netflix/genie/web/scripts/ScriptManager.java

```diff
@@ -0,0 +1,284 @@
+    private final ResourceLoader resourceLoader;
+    private final MeterRegistry meterRegistry;
+
+    /**
+     * Constructor.
+     *
+     * @param properties properties
+     * @param taskScheduler task scheduler
+     * @param executorService executor service
+     * @param scriptEngineManager script engine manager
+     * @param resourceLoader resource loader
+     * @param meterRegistry meter registry
+     */
+    public ScriptManager(
+        final ScriptManagerProperties properties,
+        final TaskScheduler taskScheduler,
+        final ExecutorService executorService,
+        final ScriptEngineManager scriptEngineManager,
+        final ResourceLoader resourceLoader,
+        final MeterRegistry meterRegistry
+    ) {
+        this.properties = properties;
+        this.taskScheduler = taskScheduler;
+        this.executorService = executorService;
+        this.scriptEngineManager = scriptEngineManager;
+        this.resourceLoader = resourceLoader;
+        this.meterRegistry = meterRegistry;
+    }
+
+    /**
+     * Start managing the given script, loading it ASAP (asynchronously) and refreshing it periodically.
+     * Because the execution of this task is asynchronous, this method returns immediately and does not surface any
+     * loading errors encountered.
+     *
+     * @param scriptUri the script to load and manage
+     */
+    public void manageScript(final URI scriptUri) {
+        final AtomicBoolean newKey = new AtomicBoolean(false);
+        final AtomicReference<CompiledScript> compiledScriptReference = this.scriptsMap.computeIfAbsent(
+            scriptUri,
+            (key) -> {
+                newKey.set(true);
+                return new AtomicReference<>();
+            }
+        );
+        if (newKey.get()) {
+            this.taskScheduler.scheduleAtFixedRate(
+                new LoadScriptTask(scriptUri, compiledScriptReference),
+                Instant.now(),
+                Duration.ofMillis(this.properties.getRefreshInterval())
+            );
+            log.debug("Scheduled periodic refresh task for script: {}", scriptUri);
+        }
+    }
+
+    /**
+     * Evaluate a given script.
+     *
+     * @param scriptUri the script URI
+     * @param bindings  the input parameter bindings
+     * @return the result of the evaluation
+     * @throws ScriptNotConfiguredException if the script is not loaded (due to invalid URI or compilation errors).
+     * @throws ScriptExecutionException     if the script evaluation produces an error
+     */
+    protected Object evaluateScript(
+        final URI scriptUri,
+        final Bindings bindings,
+        final long timeout
+    ) throws ScriptNotConfiguredException, ScriptExecutionException {
+
+        final AtomicReference<CompiledScript> compiledScriptReference = this.scriptsMap.get(scriptUri);
+
+        if (compiledScriptReference == null) {
+            throw new ScriptNotConfiguredException("Unknown script: " + scriptUri);
```

Comment:
Wouldn't it be good to wrap these in the try finally so there are metrics collected on these exceptions being thrown?